src/dune/action_dune_lang.ml

```diff
@@ -29,6 +29,47 @@ include Action_ast.Make (String_with_vars) (String_with_vars)
           (Uast)
 module Mapper = Action_mapper.Make (Uast) (Uast)
 
+let ensure_at_most_one_dynamic_run ~loc action =
```

Comment:
The reason for the limit of having at most one `Dynamic_run` per action is unclear. It would be nice to explain it here in a comment. The limit is also not mentioned in the docs for `(dynamic-run <prog> <args>)`; it probably should be.

Reply:
Added a comment.